src/nvim/eval.c

```diff
@@ -11025,39 +11055,40 @@ static void f_indent(typval_T *argvars, typval_T *rettv, FunPtr fptr)
  */
 static void f_index(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
-  list_T      *l;
-  listitem_T  *item;
   long idx = 0;
-  int ic = FALSE;
+  bool ic = false;
 
   rettv->vval.v_number = -1;
   if (argvars[0].v_type != VAR_LIST) {
     EMSG(_(e_listreq));
     return;
   }
-  l = argvars[0].vval.v_list;
+  list_T *const l = argvars[0].vval.v_list;
   if (l != NULL) {
-    item = l->lv_first;
+    listitem_T *item = tv_list_first(l);
     if (argvars[2].v_type != VAR_UNKNOWN) {
       bool error = false;
 
       // Start at specified item.  Use the cached index that tv_list_find()
       // sets, so that a negative number also works.
-      item = tv_list_find(l, tv_get_number_chk(&argvars[2], &error));
-      idx = l->lv_idx;
-      if (argvars[3].v_type != VAR_UNKNOWN) {
-        ic = tv_get_number_chk(&argvars[3], &error);
-      }
-      if (error) {
+      idx = tv_list_uidx(l, tv_get_number_chk(&argvars[2], &error));
```

Comment:
Why not only error check `tv_get_number_chk()` here and let `tv_list_find()` handle the normalization and error handling?

Reply:
And how I am to get the number to return exactly? `tv_list_find()` does not return one and I was not feeling like refactoring it to have ret_idx argument as it is useful in two places out of 18 tv_list_find is being used.